### PR TITLE
Improving azure policy

### PR DESCRIPTION
**Pull request by HRouhani**
1. Removal of MariaDB-related Tests: Two tests related to MariaDB have been removed in response to Azure's recent changes. As detailed in Azure's official announcement ([Azure Update on MariaDB](https://learn.microsoft.com/en-us/azure/mariadb/whats-happening-to-mariadb)), the support for creating new MariaDB server instances via the Azure portal ceased on January 19, 2024. While Azure CLI will continue to support MariaDB instance creation until March 19, 2024, this retirement necessitates the removal of these tests from our suite to align with the updated Azure capabilities.

2. Elimination of Redundant Test: Additionally, one test has been removed due to its redundancy. This step ensures our testing process remains efficient and focused on unique scenarios, enhancing overall test suite effectiveness.

### CODE REVIEW

File: core/mondoo-azure-security.mql.yaml [hunk 1]
```diff
@@ -491,17 +484,16 @@ queries:
     impact: 80
     mql: |
       azure.subscription.network.securityGroups {
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:486]: Suggest using the `.all()` statement as with the other PR:
```
testNSG = azure.subscription.network.securityGroup.securityRules
        .where(
          properties.access == 'Allow'
            && direction == 'Inbound'
            && properties.protocol == /TCP|\*/
            && properties.sourceAddressPrefix == /\*|0\.0\.0\.0|<nw>\/0|\/0|internet|any/
        )

      testNSG.all(properties.destinationPortRange != "*")

      a = testNSG.all(destinationPortRange.all(fromPort < 3389 && toPort < 3389))

      b = testNSG.all(destinationPortRange.all(fromPort > 3389 && toPort > 3389))

      isPortRangeValid = a || b
      isPortRangeValid
      
```

File: core/mondoo-azure-security.mql.yaml [hunk 2]
```diff
@@ -564,17 +556,16 @@ queries:
...
-        securityRules.where( properties['access'] == 'Allow' && properties['direction'] == 'Inbound' && properties['protocol'] == /TCP|\*/ && properties['sourceAddressPrefix'] == /\*|0\.0\.0\.0|<nw>\/0|\/0|internet|any/ )
-        } {
-          securityRules {
-            properties["destinationPortRange"] != "*"
-            properties["destinationPortRange"] != 3389
-            if (properties["destinationPortRange"].split("-").length > 1) {
-              a = properties["destinationPortRange"].split("-")[0] < 3389 && properties["destinationPortRange"].split("-")[1] < 3389
-              b = properties["destinationPortRange"].split("-")[0] > 3389 && properties["destinationPortRange"].split("-")[1] > 3389
-              a || b
+        securityRules.where(properties.access == 'Allow' && direction == 'Inbound' && properties.protocol == /TCP|\*/ && properties.sourceAddressPrefix == /\*|0\.0\.0\.0|<nw>\/0|\/0|internet|any/)
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:559]: Same as above.

File: core/mondoo-azure-security.mql.yaml [hunk 3]
```diff
@@ -696,7 +687,8 @@ queries:
         2. Run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.storage.accounts { containers { properties['publicAccess'] == "None" } }"
+          cnspec run azure -c "azure.subscription.storage.accounts { properties.AllowBlobPublicAccess == "false" }"
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:690]:
```suggestion
          cnspec run azure -c "azure.subscription.storage.accounts.all(properties.AllowBlobPublicAccess == "false" )"
```

File: core/mondoo-azure-security.mql.yaml [hunk 4]
```diff
@@ -696,7 +687,8 @@ queries:
...
+          cnspec run azure -c "azure.subscription.storage.accounts { containers { properties.publicAccess == "None" } }"
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:691]:
```suggestion
          cnspec run azure -c "azure.subscription.storage.accounts.all(containers.all(properties.publicAccess == "None" )) }"
```

File: core/mondoo-azure-security.mql.yaml [hunk 5]
```diff
@@ -713,7 +705,9 @@ queries:
         3. Run this query:
 
           ```mql
-          azure.subscription.storage.accounts { containers { properties['publicAccess'] == "None" } }
+          azure.subscription.storage.accounts { properties.AllowBlobPublicAccess == "false" }
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:708]: Also using `.all()` queries here for better actionable output.

File: core/mondoo-azure-security.mql.yaml [hunk 6]
```diff
@@ -763,7 +757,7 @@ queries:
         2. Run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.storage.accounts { containers { properties['publicAccess'] == "None" } }"
+          cnspec run azure -c "azure.subscription.storage.accounts {properties.NetworkRuleSet.defaultAction == "Deny"}"
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:760]: Update to `.all()`

File: core/mondoo-azure-security.mql.yaml [hunk 7]
```diff
@@ -780,7 +774,7 @@ queries:
         3. Run this query:
 
           ```mql
-          azure.subscription.storage.accounts { containers { properties['publicAccess'] == "None" } }
+          azure.subscription.storage.accounts {properties.NetworkRuleSet.defaultAction == "Deny"}
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:777]: same

File: core/mondoo-azure-security.mql.yaml [hunk 8]
```diff
@@ -826,7 +820,7 @@ queries:
         2. Run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.storage.accounts.all ( properties['networkAcls']['bypass'] == "AzureServices")"
+          cnspec run azure -c "azure.subscription.storage.accounts {properties.NetworkRuleSet.bypass.contains("AzureServices")}"
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:823]: also using `.all()` with new v9

File: core/mondoo-azure-security.mql.yaml [hunk 9]
```diff
@@ -930,7 +870,7 @@ queries:
         2. Run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.sql.servers { auditingPolicy['retentionDays'] }"
+          cnspec run azure -c "azure.subscription.sql.servers {auditingPolicy.retentionDays >= 30}"
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:873]: same

File: core/mondoo-azure-security.mql.yaml [hunk 10]
```diff
@@ -947,7 +887,7 @@ queries:
         3. Run this query:
 
           ```mql
-          azure.subscription.sql.servers { auditingPolicy['retentionDays'] }
+          azure.subscription.sql.servers { auditingPolicy.retentionDays }
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:890]: same

File: core/mondoo-azure-security.mql.yaml [hunk 11]
```diff
@@ -1094,7 +1013,7 @@ queries:
         To audit Microsoft Azure with `cnspec run`, run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.web.apps { identity["type"] == "SystemAssigned" || identity["principalId"] != null }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          cnspec run azure -c "azure.subscription.web.apps { identity.type == "SystemAssigned" || identity.principalId != empty }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1016]: same

File: core/mondoo-azure-security.mql.yaml [hunk 12]
```diff
@@ -1112,7 +1031,7 @@ queries:
         2. Run this query:
 
           ```mql
-          azure.subscription.web.apps { identity["type"] == "SystemAssigned" || identity["principalId"] != null }
+          azure.subscription.web.apps { identity.type == "SystemAssigned" || identity.principalId != empty }
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1034]: same

File: core/mondoo-azure-security.mql.yaml [hunk 13]
```diff
@@ -1140,7 +1057,7 @@ queries:
         To audit Microsoft Azure with `cnspec run`, run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.web.apps { configuration.properties["minTlsVersion"] == "1.2" }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          cnspec run azure -c "azure.subscription.web.apps { configuration.properties.minTlsVersion == "1.2" }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1060]: same

File: core/mondoo-azure-security.mql.yaml [hunk 14]
```diff
@@ -1158,7 +1075,7 @@ queries:
         2. Run this query:
 
           ```mql
-          azure.subscription.web.apps { configuration.properties["minTlsVersion"] == "1.2" }
+          azure.subscription.web.apps { configuration.properties.minTlsVersion == "1.2" }
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1078]: same

File: core/mondoo-azure-security.mql.yaml [hunk 15]
```diff
@@ -1248,7 +1161,7 @@ queries:
         To audit Microsoft Azure with `cnspec run`, run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.keyVault.vaults { properties["enablePurgeProtection"] == "true" properties["enableSoftDelete"] == "true" }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          cnspec run azure -c "azure.subscription.keyVault.vaults { properties.enablePurgeProtection == "true" properties.enableSoftDelete == "true" }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1164]: same

File: core/mondoo-azure-security.mql.yaml [hunk 16]
```diff
@@ -1266,7 +1179,7 @@ queries:
         2. Run this query:
 
           ```mql
-          azure.subscription.keyVault.vaults { properties["enablePurgeProtection"] == "true" properties["enableSoftDelete"] == "true" }
+          azure.subscription.keyVault.vaults { properties.enablePurgeProtection == "true" properties.enableSoftDelete == "true" }
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1182]: same

File: core/mondoo-azure-security.mql.yaml [hunk 17]
```diff
@@ -1297,7 +1208,7 @@ queries:
         To audit Microsoft Azure with `cnspec run`, run this query:
 
           ```bash
-          cnspec run azure -c "azure.subscription.keyVault.vaults { diagnosticSettings.any( properties["logs"].where( _["category"] == "AuditEvent" && _["retentionPolicy"]["days"] >= 180 )  ) }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          cnspec run azure -c "azure.subscription.keyVault.vaults { diagnosticSettings.any( properties.logs.where(category == "AuditEvent" && retentionPolicy.days >= 180 )  ) }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1211]: same

**Review by mm-weber — CHANGES_REQUESTED** [review]
Good stuff @HRouhani .

Besides the one NSG query we're currently working on, there's a few times where the `.all()` isn't reflected in the descriptions, yet.

File: core/mondoo-azure-security.mql.yaml (hunk 1 above)
Comment (HRouhani) [reply core/mondoo-azure-security.mql.yaml:486]: Thank you for the suggestion. However, it appears that it doesn't effectively address all the test scenarios we have. I appreciate your input and look forward to exploring alternative solutions or Keeping the Original one that might be more comprehensive.

File: core/mondoo-azure-security.mql.yaml (hunk 2 above)
Comment (HRouhani) [reply core/mondoo-azure-security.mql.yaml:559]: Thank you for the suggestion. However, it appears that it doesn't effectively address all the test scenarios we have. I appreciate your input and look forward to exploring alternative solutions or Keeping the Original one that might be more comprehensive.

File: core/mondoo-azure-security.mql.yaml (hunk 3 above)
Comment (HRouhani) [reply core/mondoo-azure-security.mql.yaml:690]: I intentionally left the description as-is to facilitate a better understanding for the community. This approach should help them grasp how to test effectively and enable more informed comparisons. What are your thoughts on this strategy?

File: core/mondoo-azure-security.mql.yaml (hunk 5 above)
Comment (HRouhani) [reply core/mondoo-azure-security.mql.yaml:708]: I intentionally left the description as-is to facilitate a better understanding for the community. This approach should help them grasp how to test effectively and enable more informed comparisons. What are your thoughts on this strategy?

File: core/mondoo-azure-security.mql.yaml [hunk 18]
```diff
@@ -1996,49 +1555,20 @@ queries:
...
+            && properties.protocol == /UDP/
+            && properties.sourceAddressPrefix == /\*|0\.0\.0\.0|<nw>\/0|\/0|internet|any/
+        )
+        )
+        allNsgUDP.all(properties.destinationPortRange != "*")
+        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 53 && toPort >= 53)))
+        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 123 && toPort >= 123)))
+        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 161 && toPort >= 161)))
+        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 389 && toPort >= 389)))
+        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 1900 && toPort >= 1900)))" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>```
```
Comment (mm-weber) [inline core/mondoo-azure-security.mql.yaml:1571]:
```suggestion
        allNsgUDP.all(securityRules.none(destinationPortRange.any(fromPort <= 1900 && toPort >= 1900)))```
```

**Review by mm-weber — CHANGES_REQUESTED** [review]
Good stuff.
LGTM<s>, just commit that little typo, and we're good.</s>
(My bad... good to go)

**Review by mm-weber — APPROVED** [review]
LGTM.
Good work @HRouhani

**Pull request merged**